Modules/_ctypes/cfield.c

```diff
@@ -398,35 +343,9 @@ Py_ssize_t NUM_BITS(Py_ssize_t bitsize) {
 /* This macro RETURNS the first parameter with the bit field CHANGED. */
 #define SET(type, x, v, size)                                                 \
     (NUM_BITS(size) ?                                                   \
-     ( ( (type)x & ~(BIT_MASK(type, size) << LOW_BIT(size)) ) | ( ((type)v & BIT_MASK(type, size)) << LOW_BIT(size) ) ) \
+     ( ( (type)(x) & ~(BIT_MASK(type, size) << LOW_BIT(size)) ) | ( ((type)(v) & BIT_MASK(type, size)) << LOW_BIT(size) ) ) \
```

Comment:
Can't comment here but you also need wrapping in `(type)v`.

Reply:
Right, thanks.
(There are other possible improvements here, but I might be able to get rid of the macro entirely. Here I only want to make new usages safer.)